src/main/java/org/openrewrite/java/spring/util/MemberReferenceToMethodInvocation.java

```diff
@@ -37,11 +38,23 @@ public J visitMemberReference(J.MemberReference memberRef, ExecutionContext ctx)
         }
 
         List<Param> args = getLambdaArgNames(mr.getMethodType());
-        String templateCode = String.format("(%s) -> %s.%s(%s)",
-                args.stream().map(Param::toString).collect(joining(", ")),
-                mr.getContaining(),
-                mr.getReference().getSimpleName(),
-                args.stream().map(Param::getName).collect(joining(", ")));
+        String templateCode;
+        if (!mr.getMethodType().getParameterTypes().isEmpty()) {
+            templateCode = String.format("(%s) -> %s.%s(%s)",
+                    args.stream().map(Param::toString).collect(joining(", ")),
+                    mr.getContaining(),
+                    mr.getReference().getSimpleName(),
+                    args.stream().map(Param::getName).collect(joining(", ")));
+        } else if (mr.getContaining() instanceof J.Identifier && ("this".equals(((J.Identifier) mr.getContaining()).getSimpleName()) || "super".equals(((J.Identifier) mr.getContaining()).getSimpleName()))) {
+            templateCode = String.format("() -> %s.%s()",
+                    mr.getContaining(),
+                    mr.getReference().getSimpleName());
+        } else {
+            templateCode = String.format("(%s) -> %s.%s()",
+                    args.stream().map(Param::toString).collect(joining(", ")),
+                    args.get(0).getName(),
+                    mr.getReference().getSimpleName());
+        }
         return JavaTemplate.builder(templateCode)
                 .contextSensitive()
```

Comment:
Would be interesting to see if we could use `bindType()` here instead:
```suggestion
                .bindType(TypeUtils.toString(mr.getType()))
```

Reply:
This creates failing tests in `MigrateAuditorAwareToOptionalTest` due to missing type information. 

`afterTypeValidationOptions(TypeValidation.none())` fixes these tests, but I think we'd rather keep the type info?

Reply:
Indeed. These tests fail because the method reference includes `this` somewhere in its `containing`. So it doesn't look like this will be applicable here.